bazel_scripts/distribution_config.bzl

```diff
@@ -0,0 +1,88 @@
+"""Configuration for building Enso Engine distribution."""
+
+load("//toolchains/sbt:run_sbt.bzl", "run_sbt")
+
+def _sbt_path_props(base, sep):
+    """Generates SBT path properties for a given base variable and separator."""
+    return [
+        "-Dsbt.repositories.local={}{}.sbt{}repositories".format(base, sep, sep),
+        "-Dsbt.global.base={}".format(base),
+        "-Dsbt.boot.directory={}{}.sbt{}boot".format(base, sep, sep),
+        "-Dsbt.ivy.home={}{}.ivy2".format(base, sep),
+        "-Dsbt.coursier.home={}{}.coursier".format(base, sep),
+    ]
+
+_SBT_PATH_PROPS = select({
+    "@platforms//os:windows": _sbt_path_props("%TEMP%", "\\"),
+    "//conditions:default": _sbt_path_props("$$TMPDIR", "/"),
+})
+
+_COMMON_JVM_OPTS = [
+    "-Xss16M",
+    "-Xmx4G",
```

Comment:
That feels... low? 4GB was for SBT itself I believe. Native Image itself requires > 10GB